integration/benchmark/worker_stats_tracker.go

```diff
@@ -1,48 +1,83 @@
 package benchmark
 
 import (
-	"fmt"
+	"context"
 	"sync"
 	"time"
 
-	"github.com/jedib0t/go-pretty/table"
+	"github.com/VividCortex/ewma"
+	"github.com/rs/zerolog"
 )
 
-type WorkerStats struct {
-	workers     int
-	txsSent     int
-	txsExecuted int
+type workerStats struct {
+	workers         int
+	txsSent         int
+	txsExecuted     int
+	txsSentEWMA     float64
+	txsExecutedEWMA float64
 }
 
 // WorkerStatsTracker keeps track of worker stats
 type WorkerStatsTracker struct {
-	mux              sync.Mutex
-	stats            WorkerStats
-	txsSentPerSecond map[int64]int // tracks txs sent at the timestamp in seconds
-
-	printer *Worker
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+
+	mux             sync.Mutex
+	stats           workerStats
+	txsSentEWMA     ewma.MovingAverage
+	txsExecutedEWMA ewma.MovingAverage
 }
 
+const defaultEWMAAge = 10
+
 // NewWorkerStatsTracker returns a new instance of WorkerStatsTracker
-func NewWorkerStatsTracker() *WorkerStatsTracker {
-	return &WorkerStatsTracker{
-		txsSentPerSecond: make(map[int64]int),
+func NewWorkerStatsTracker(ctx context.Context) *WorkerStatsTracker {
+	ctx, cancel := context.WithCancel(ctx)
+	st := &WorkerStatsTracker{
+		ctx:             ctx,
+		cancel:          cancel,
+		txsSentEWMA:     ewma.NewMovingAverage(defaultEWMAAge),
+		txsExecutedEWMA: ewma.NewMovingAverage(defaultEWMAAge),
 	}
+
+	st.wg.Add(1)
+	go st.updateEWMAforever()
+	return st
 }
 
-// StartPrinting starts reporting of worker stats
-func (st *WorkerStatsTracker) StartPrinting(interval time.Duration) {
-	printer := NewWorker(0, interval, func(_ int) { fmt.Println(st.Digest()) })
-	st.printer = printer
-	st.printer.Start()
+func (st *WorkerStatsTracker) updateEWMAforever() {
+	defer st.wg.Done()
+
+	t := time.NewTicker(time.Second)
+	defer t.Stop()
+
+	lastStats := st.getStats()
+	for {
+		select {
+		case <-t.C:
+			stats := st.getStats()
+			st.updateEWMAonce(lastStats, stats)
+			lastStats = stats
+		case <-st.ctx.Done():
+			return
+		}
+	}
 }
 
-// StopPrinting stops reporting of worker stats
-func (st *WorkerStatsTracker) StopPrinting() {
-	st.printer.Stop()
+func (st *WorkerStatsTracker) updateEWMAonce(lastStats, stats workerStats) {
+	st.mux.Lock()
+	defer st.mux.Unlock()
+
+	st.txsSentEWMA.Add(float64(stats.txsSent - lastStats.txsSent))
```

Comment:
can you add comments here explaining what's being tracked? Looks like we're keeping track of the transactions sent since the last time